src/configure/helper/templateHelper.ts

```diff
@@ -112,35 +139,50 @@ export async function analyzeRepoAndListAppropriatePipeline(repoPath: string, re
     templateResult = targetResource && !!targetResource.kind ? templateResult.filter((template) => !template.targetKind || template.targetKind.toLowerCase() === targetResource.kind.toLowerCase()) : templateResult;
     templateResult = templateResult.filter((pipelineTemplate) => pipelineTemplate.enabled);
 
-    // remove duplicate named template:
-    templateResult = removeDuplicates(templateResult);
     return templateResult;
-
 }
 
-export async function analyzeRepoAndListAppropriatePipeline2(azureSession: AzureSession, repoPath: string, repositoryProvider: RepositoryProvider, repoAnalysisParameters: RepositoryAnalysisParameters, targetResource?: GenericResource): Promise<PipelineTemplateMetadata[]> {
+export async function analyzeRepoAndListAppropriatePipeline2(azureSession: AzureSession, repoPath: string, repositoryProvider: RepositoryProvider, repoAnalysisParameters: RepositoryAnalysisParameters, targetResource?: GenericResource): Promise<PipelineTemplate[]> {
 
-    //TO:DO - Merge local repo analysis (Some changes in the definition of AnalysisResult required)
-    let templateResult: PipelineTemplateMetadata[] = [];
+    var pipelineTemplates: PipelineTemplate[] = [];
+    var localPipelineTemplatesPromise: Promise<LocalPipelineTemplate> = await this.analyzeRepoAndListAppropriatePipeline(repoPath, repositoryProvider, repoAnalysisParameters);
 
     let serviceClient = new TemplateServiceClient(azureSession.credentials);
-    templateResult = await serviceClient.getTemplates(repoAnalysisParameters);
-    templateResult = templateResult.sort((a, b) => {
-        if (a.templateWeight > b.templateWeight) { return 1; }
-        else { return -1; }
-    });
-    return templateResult;
+    var templateResultPromise = await serviceClient.getTemplates(repoAnalysisParameters);
+
+    return Q.all([localPipelineTemplatesPromise, templateResultPromise])
```

Comment:
We should add fallback logic to return local templates if we fail to get remote templates.

Reply:
+1

Reply:
Should I go back to the previous design where we will have a call made for local templates, and the condition of templateServiceEnabled will be used to determine whether we need to have a call for remote templates?

Reply:
@bishal-pdMSFT will templateService only return templates for github workflow? Are we planning to add AzurePipelines templates in template service?

Reply:
Code updated. Even if the call to remote templates fails due to any reason, it will still result the list of local templates.

Reply:
@anuragc617 right now there is no plan to add azure pipelines templates. But lets not take dependency on this as things might change